test/unit/rh_cloud_http_proxy_test.rb

```diff
@@ -0,0 +1,64 @@
+require 'test_plugin_helper'
+
+class RhCloudHttpProxyTest < ActiveSupport::TestCase
+  setup do
+    @global_content_proxy_mock = 'http://global:content@localhost:8888'
+    @global_foreman_proxy_mock = 'http://global:foreman@localhost:8888'
+    @katello_cdn_proxy_mock = {
+      host: 'localhost',
+      port: '8888',
+      user: 'katello',
+      password: 'c@n',
+      scheme: 'http',
+    }
+  end
+
+  test 'selects global content proxy' do
+    setup_global_content_proxy
+    setup_global_foreman_proxy
+    setup_cdn_proxy do
+      assert_equal @global_content_proxy_mock, ForemanRhCloud.proxy_setting
+    end
+  end
+
+  test 'selects cdn proxy' do
+    setup_global_foreman_proxy
+    setup_cdn_proxy do
+      assert_equal @katello_cdn_proxy_mock, ForemanRhCloud.proxy_setting
```

Comment:
You have to construct the string here. `@katello_cdn_proxy_mock` is a hash object.